templates/redis/bin/common.sh

```diff
@@ -0,0 +1,85 @@
+# Environment variable common to all scripts
+APISERVER=https://kubernetes.default.svc
+SERVICEACCOUNT=/var/run/secrets/kubernetes.io/serviceaccount
+NAMESPACE=$(cat ${SERVICEACCOUNT}/namespace)
+TOKEN=$(cat ${SERVICEACCOUNT}/token)
+CACERT=${SERVICEACCOUNT}/ca.crt
+
+TIMEOUT=3
+
+PODNAME=$(echo $HOSTNAME | cut -d. -f1,2)
+PODIP=$(grep "${PODNAME}" /etc/hosts | cut -d$'\t' -f1)
+# redis-redis-0.redis-redis.openstack.svc.cluster.local
+PODFQDN=$(getent hosts "${PODIP}" | awk '{print $2}')
+# redis-redis.openstack.svc.cluster.local
+SVCFQDN=$(echo "${PODFQDN}" | sed 's/^[^\.]*.//')
+
+function log() {
+    echo "$(date +%F_%H_%M_%S) $*"
+}
+
+function log_error() {
+    echo "$(date +%F_%H_%M_%S) ERROR: $*"
+}
+
+function generate_configs() {
+    # Copying config files except template files
+    mkdir -p /var/lib/kolla/config_files/generated
+    tar -C /var/lib/kolla/config_files --exclude '..*' --exclude '*.in' -h -c src | tar -C /var/lib/kolla/config_files/generated -x --strip=1
+    # Generating config files from templates
+    cd /var/lib/kolla/config_files/src
+    for cfg in $(find -L * -name '*.conf.in'); do
+        log "Generating config file from template $PWD/${cfg}"
+        sed -e "s/{ PODNAME }/${PODNAME}/" -e "s/{ PODIP }/${PODIP}/" -e "s/{ PODFQDN }/${PODFQDN}/" "${cfg}" > "/var/lib/kolla/config_files/generated/${cfg%.in}"
+    done
+}
+
+function is_bootstrap_pod() {
+    echo "$1" | grep -qe '-0$'
+}
+
+function extract() {
+    local var="$1"
+    local output="$2"
+    # parse curl vars as well as kube api error fields
+    echo "$output" | awk -F'[:,]' "/\"?${var}\"?:/ {print \$2; exit}"
+}
+
+function configure_pod_label() {
+    local pod="$1"
+    local patch="$2"
+    local success="$3"
+    local curlvars="\nexitcode:%{exitcode}\nerrormsg:%{errormsg}\nhttpcode:%{response_code}\n"
+
+    response=$(curl -s -w "${curlvars}" --cacert ${CACERT} --header "Content-Type:application/json-patch+json" --header "Authorization: Bearer ${TOKEN}" --request PATCH --data "$patch" ${APISERVER}/api/v1/namespaces/${NAMESPACE}/pods/${pod})
```

Comment:
when we add tls support the CACERT should be the system wide one.

Reply:
This one is the CA injected by kubernetes to target the API server, I think those are different from the one we store system wide, right?

Reply:
the bundle we create in the openstack-operator will also have the cluster CA, but I think for this use case it is ok to use the one coming with the serviceaccount information